pkg/controller/common/helpers.go

```diff
@@ -271,9 +296,107 @@ func IgnParseWrapper(rawIgn []byte) (ign2types.Config, error) {
 	return ign2types.Config{}, errors.Errorf("parsing Ignition config spec v2 failed with error: %v\nReport: %v", err, rpt)
 }
 
+// ParseAndConvertConfig parses rawIgn for both V2 and V3 ignition configs and returns
+// a V3 or an error.
+func ParseAndConvertConfig(rawIgn []byte) (ign3types.Config, error) {
+	ignconfigi, err := IgnParseWrapper(rawIgn)
+	if err != nil {
+		return ign3types.Config{}, errors.Wrapf(err, "failed to parse Ignition config")
+	}
+
+	switch typedConfig := ignconfigi.(type) {
+	case ign3types.Config:
+		return ignconfigi.(ign3types.Config), nil
+	case ign2types.Config:
+		ignconfv2 := removeIgnDuplicateFilesAndUnits(ignconfigi.(ign2types.Config))
+		convertedIgnV3, err := convertIgnition2to3(ignconfv2)
+		if err != nil {
+			return ign3types.Config{}, errors.Wrapf(err, "failed to convert Ignition config spec v2 to v3")
+		}
+		return convertedIgnV3, nil
+	default:
+		return ign3types.Config{}, errors.Errorf("unexpected type for ignition config: %v", typedConfig)
+	}
+}
+
+// Function to remove duplicated files/units from a V2 MC, since the translator
+// (and ignition spec V3) does not allow for duplicated entries in one MC.
+// This should really not change the actual final behaviour, since it keeps
+// ordering into consideration and has contents from the highest alphanumeric
+// MC's final version of a file.
+// Note:
+// Append is not considered since we do not allow for appending
+// Units have one exception: dropins are concat'ed
+
+func removeIgnDuplicateFilesAndUnits(ignConfig ign2types.Config) ign2types.Config {
```

Comment:
I think this function would benefit from a couple of unit tests scenarios 👼 just to make sure I don't go and change something and break some cases - what do you think?

Reply:
Added a unit test, thanks!